pkg/assume/assume.go

```diff
@@ -451,16 +451,29 @@ func AssumeCommand(c *cli.Context) error {
 		if assumeFlags.String("exec") != "" {
 			return RunExecCommandWithCreds(assumeFlags.String("exec"), creds, region)
 		}
+
 		// DO NOT REMOVE, this interacts with the shell script that wraps the assume command, the shell script is what configures your shell environment vars
 		// to export more environment variables, add then in the assume and assume.fish scripts then append them to this output preparation function
 		// the shell script treats "None" as an empty string and will not set a value for that positional output
+
+		// If the profile uses "credential_process" to source credential externally then do not set accessKeyId, secretAccessKey, sessionToken
+		// so that aws cli automatically refreshes credential when they expire.
+		if profile.ProfileType == "AWS_CREDENTIAL_PROCESS" {
+			output := PrepareStringsForShellScript([]string{"", "", "", "", region, sessionExpiration, profile.AWSConfig.SSOStartURL, profile.AWSConfig.SSORegion, profile.AWSConfig.SSORoleName, profile.AWSConfig.SSORegion, profile.AWSConfig.SSOAccountID})
```

Comment:
In this code I don't think you should use returning the values for ssoStartURL etc as per the output when teh sso flag is used.

The reason we return these for the sso flag is because the SSO flag is used when you don;t have a profile configured already, and these flags help us to re assume the same config again later.

for profiles with a credential process, you should simply do the following

```suggestion
			output := PrepareStringsForShellScript([]string{"", "", "", profile.Name, region, "", "", "", "", "", ""})
```